cmd/query/app/static_handler.go

```diff
@@ -94,11 +126,54 @@ func NewStaticAssetsHandler(staticAssetsRoot string, options StaticAssetsHandler
 		}
 		indexBytes = basePathPattern.ReplaceAll(indexBytes, []byte(fmt.Sprintf(basePathReplace, options.BasePath)))
 	}
-	return &StaticAssetsHandler{
-		options:   options,
-		indexHTML: indexBytes,
-		assetsFS:  assetsFS,
-	}, nil
+
+	return indexBytes, nil
+}
+
+func (sH *StaticAssetsHandler) watch() {
+	if sH.options.UIConfigPath == "" {
+		return
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		sH.options.Logger.Error("failed to create a new watcher for the UI config", zap.Error(err))
+		return
+	}
+	defer watcher.Close()
+
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Op&fsnotify.Write == fsnotify.Write {
+					sH.options.Logger.Info("reloading UI config", zap.String("filename", sH.options.UIConfigPath))
+
+					sH.contentMutex.Lock()
+					sH.indexHTML, err = loadIndexBytes(sH.assetsFS.Open, sH.options)
+					sH.contentMutex.Unlock()
+					if err != nil {
+						sH.options.Logger.Error("error while reloading the UI config", zap.Error(err))
+					}
+
+				}
+			case err, ok := <-watcher.Errors:
+				if !ok {
+					return
+				}
+				sH.options.Logger.Error("event", zap.Error(err))
+			}
+		}
+	}()
+
+	err = watcher.Add(sH.options.UIConfigPath)
```

Comment:
what is this trying to achieve?

Reply:
This specific line? That's how we tell the watcher which file we are interested in.